src/Adapter/Product/Combination/CommandHandler/GenerateCombinationsHandler.php

```diff
@@ -0,0 +1,165 @@
+
+use Combination;
+use Db;
+use PrestaShop\PrestaShop\Adapter\Product\AbstractProductHandler;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Command\GenerateCombinationsCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\CommandHandler\GenerateCombinationsHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Exception\CombinationException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\ValueObject\CombinationId;
+use PrestaShop\PrestaShop\Core\Product\Generator\CombinationGeneratorInterface;
+use PrestaShop\PrestaShop\Core\Product\Generator\GeneratedCombination;
+use PrestaShopException;
+use Product;
+use SpecificPriceRule;
+
+final class GenerateCombinationsHandler extends AbstractProductHandler implements GenerateCombinationsHandlerInterface
+{
+    /**
+     * @var CombinationGeneratorInterface
+     */
+    private $combinationGenerator;
+
+    /**
+     * @param CombinationGeneratorInterface $combinationGenerator
+     */
+    public function __construct(CombinationGeneratorInterface $combinationGenerator)
+    {
+        $this->combinationGenerator = $combinationGenerator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(GenerateCombinationsCommand $command): array
+    {
+        $product = $this->getProduct($command->getProductId());
+        $attributesByGroup = $command->getAttributesByGroup();
+
+        SpecificPriceRule::disableAnyApplication();
+
+        $generatedCombinations = $this->combinationGenerator->bulkGenerate($attributesByGroup);
+        $combinationIds = $this->addToDatabase($generatedCombinations, $product);
+
+        Product::updateDefaultAttribute($product->id);
+        SpecificPriceRule::enableAnyApplication();
+        SpecificPriceRule::applyAllRules([(int) $product->id]);
+
+        return $combinationIds;
+    }
+
+    /**
+     * @param GeneratedCombination[] $generatedCombinations
+     * @param Product $product
+     *
+     * @return CombinationId[]
+     */
+    private function addToDatabase(array $generatedCombinations, Product $product): array
+    {
+        $addedCombinationsIds = [];
+        foreach ($generatedCombinations as $generatedCombination) {
+            // checks if combination already exist. @todo: should be possible to optimize this check.
+            $existingCombinationId = $product->productAttributeExists(
+                $generatedCombination->getAttributeIds(),
+                false,
+                null,
+                true,
+                true
+            );
+
+            if ($existingCombinationId) {
+                //skip if combination already exists
+                continue;
+            }
+
+            $newCombination = new Combination();
+            $newCombination->id_product = $product->id;
+            $newCombination->default_on = 0;
+            $product->setAvailableDate();
+
+            $dbInstance = Db::getInstance();
```

Comment:
@zuk3975 this is command, shouldn't we have instance of db in constructor? command has only one job, it's weird having to pass this instance between methods

Reply:
yeah I agree its a bit weird :sweat_smile: Will change once I know if this POC is promising. (now it seems it might not see the daylight at all)